presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlockBuilder.java

```diff
@@ -300,8 +300,18 @@ public Block copyRegion(int positionOffset, int length)
         }
 
         int[] newOffsets = Arrays.copyOfRange(offsets, positionOffset, positionOffset + length + 1);
+        // set new offsets to start from beginning of slice (since we are copying)
+        for (int i = 0; i < newOffsets.length; i++) {
+            newOffsets[i] -= offsets[positionOffset];
+        }
```

Comment:
I would use `int[] newOffsets = new int[...]` instead of `Arrays.copyOfRange`

Reply:
I use the common method `sliceOffsets` now